src/app/[locale]/seller/registration/page.tsx

```diff
@@ -57,59 +57,55 @@ const SellerRegistrationForm = () => {
   const [linkUrl, setLinkUrl] = useState('');
 
   useEffect(() => {
-    const fetchSellCenter = async () => {
+    const fetchData = async () => {
+      if (!currentUser) {
+        logger.info('User not logged in; attempting auto-login..');
+        autoLoginUser();
+        return;
+      }
+  
       try {
-        const mapCenterData = await fetchMapCenter(); // Function to fetch map center from backend
-        if (mapCenterData) {
-          const { longitude, latitude } = mapCenterData;
-          if (longitude !== undefined && latitude !== undefined) {
-            setSellCenter({ lng: longitude, lat: latitude });
+        // Fetch Sell Center
+        if (!sellCenter) {
+          const mapCenterData = await fetchMapCenter();
```

Comment:
multiple API calls (i.e. fetchMapCenter(), fetchsellerRegistration(), fetchUserSettings() ) in a single screen will increase the load time

Reply:
besides why do you need to fetch Map center here

Reply:
Some of the code is remanence of old implementation, thank you for pointing it out. I went ahead and completely separate the fetching logic, this ensures the fetch function only runs when currentUser changes.